reader_permit.hh

```diff
@@ -164,6 +164,10 @@ public:
 
     void set_timeout(db::timeout_clock::time_point timeout) noexcept;
 
+    const tracing::trace_state_ptr& trace_state() const noexcept;
+
+    void set_trace_state(tracing::trace_state_ptr trace_ptr) noexcept;
+
```

Comment:
What happens if a read starts with one trace, generates a page, then transfers into another page? We'll get mixed traces no?